code/modules/mod/mod_ai.dm

```diff
@@ -0,0 +1,77 @@
+/obj/item/mod/control/transfer_ai(interaction, mob/user, mob/living/silicon/ai/intAI, obj/item/aicard/card)
+	. = ..()
+	if(!.)
+		return
+	if(!open) //mod must be open
+		to_chat(user, "<span class='warning'>[src] must be open in order to allow a transfer.</span>")
+		return
+	switch(interaction)
+		if(AI_TRANS_TO_CARD)
+			if(!ai)
+				to_chat(user, "<span class='warning'>No AI detected in [src].</span>")
+				return
+			if(!do_after(user, 5 SECONDS, target = src))
+				return
+			intAI = ai
+			intAI.ai_restore_power()//So the AI initially has power.
+			intAI.control_disabled = TRUE
+			intAI.radio_enabled = FALSE
+			intAI.disconnect_shell()
+			intAI.forceMove(card)
+			card.AI = intAI
+			for(var/datum/action/action as anything in actions)
+				if(action.owner == intAI)
+					action.Remove(intAI)
+					qdel(action)
+			intAI.controlled_equipment = null
+			intAI.remote_control = null
+			to_chat(intAI, "<span class='notice'>You have been downloaded to a mobile storage device. Wireless connection offline.</span>")
+			to_chat(user, "<span class='boldnotice'>Transfer successful</span>: [intAI.name] ([rand(1000,9999)].exe) removed from [name] and stored within local memory.")
+			ai = null
+
+		if(AI_TRANS_FROM_CARD) //Using an AI card to upload to the suit.
+			intAI = card.AI
+			if(!intAI)
+				to_chat(user, "<span class='warning'>There is no AI currently installed on this device.</span>")
+				return
+			if(intAI.deployed_shell) //Recall AI if shelled so it can be checked for a client
+				intAI.disconnect_shell()
+			if(intAI.stat || !intAI.client)
+				to_chat(user, "<span class='warning'>[intAI.name] is currently unresponsive, and cannot be uploaded.</span>")
+				return
+			if(!do_after(user, 5 SECONDS, target = src))
```

Comment:
can you try to upload an AI to two different modular suits at once?

what if the suit closes during this do_after() (or the one from the transfering an AI from a suit to a card code)?

Reply:
can you try to upload multiple AIs to the same suit?

Reply:
do afters cancel when the tihng is moved away